### ISSUE TITLE
num_samples_processed of learning rate annealing

### ISSUE THREAD

**Issue by yinyunfeng** [PaddlePaddle/Paddle#4305]
有个疑问， 当多节点训练时，学习率退火中的num_samples_processed是一个节点的处理的样本数还是所有节点处理的样本数之和？

**Comment by typhoonzero** [comment]
您好，欢迎关注PaddlePaddle！

多节点训练时，paddle在参数优化时，主要有两种方式：

- 发送梯度到pserver进行参数优化
- trainer本地参数优化，然后将参数的diff发送到pserver

第一种方式目前应该只有momentum是这样，支持learning_rate decay的参数优化方式目前都是本地计算diff发送。所以`num_samples_processed`是一个节点的。

参考：

https://github.com/PaddlePaddle/Paddle/issues/2347

https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/trainer/RemoteParameterUpdater.cpp#L204

**Comment by Yancey1989** [comment]
由于此issue长时间没有更新，所以先关掉了，如有问题欢迎随时打开。
